modules/tinymce/src/plugins/template/main/ts/ui/Dialog.ts

```diff
@@ -110,15 +109,16 @@ const open = (editor: Editor, templateList: ExternalTemplate[]): void => {
   };
 
   const getTemplateContent = (t: InternalTemplate) => new Promise<string>((resolve, reject) => {
-    t.value.url.fold(() => resolve(t.value.content.getOr('')), (url) => XHR.send({
-      url,
-      success: (html: string) => {
-        resolve(html);
-      },
-      error: (e) => {
-        reject(e);
-      }
-    }));
+    t.value.url.fold(
+      () => resolve(t.value.content.getOr('')),
+      (url) => window.fetch(url).then((res) => {
+        if (res.ok) {
+          return resolve(res.text());
+        } else {
+          return reject();
+        }
+      }).catch(reject)
+    );
   });
```

Comment:
This can be re-written to remove the `Promise` wrapper now. For example:
```suggestion
  const getTemplateContent = (t: InternalTemplate) =>
    t.value.url.fold(
      () => Promise.resolve(t.value.content.getOr('')),
      (url) => window.fetch(url).then((res) => res.ok ? res.text() : Promise.reject())
    );
```

Reply:
Fixed